server/src/views/frame.js

```diff
@@ -125,13 +225,21 @@ exports.Frame = React.createClass({
       previousClip = null,
       nextClip = null;
 
-    for (let i = 0; i < clipNames.length; i++) {
-      let name = clipNames[i],
-        clip = shot.getClip(name);
+    for (let i=0; i<clipNames.length; i++) {
+      let clipId = clipNames[i];
+      let clip = shot.getClip(clipId);
+      let nextClip = null;
+      if (i + 1 < clipNames.length) {
+        nextClip = shot.getClip(clipNames[i+1]);
+      }
+      let previousClip = null;
+      if (i > 0) {
+        previousClip = shot.getClip(clipNames[i-1]);
+      }
 
-      snippets.push(<Snippet key={ name } clip={ clip } linkify={ this.props.linkify } shotId={ shotId } shotDomain={ shotDomain } previousClip={ previousClip } nextClip={ nextClip } />);
+      snippets.push(<Snippet staticLink={this.props.staticLink} key={ clipId } clip={ clip }  shotId={ shotId } shotDomain={ shotDomain } previousClip={ previousClip } nextClip={ nextClip } />);
 
-      if (query.clip === name) {
+      if (query.clip === clipId) {
```

Comment:
I realized I forgot to generally restore the functionality of the loaded page generally, this included.